hcloud/resource_hcloud_reversedns.go

```
@@ -0,0 +1,241 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+func resourceReverseDNS() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDNSRead,
+		Create: resourceReverseDNSCreate,
+		Delete: resourceReverseDNSDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns_ptr": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceReverseDNSRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ctx := context.Background()
+
+	ipAddress := string(d.Get("ip_address").(string))
+	rdnsType := ""
+	id, ok := d.GetOk("server_id")
+	if ok == false {
+		id, ok = d.GetOk("floating_ip_id")
+		switch {
+		case ok == false:
+			log.Printf("[WARN] Invalid id (%s), removing from state: %v", d.Id(), ok)
+			d.SetId("")
+			return nil
+		case ok == true:
+			rdnsType = "floating_ip"
+		}
+	} else {
+		rdnsType = "server"
+	}
+	ip := net.ParseIP(d.Get("ip_address").(string))
+	if ip == nil {
+		log.Printf("[WARN] The ip you provide (%s) is not valid, removing from state", ipAddress)
```

Please return an error to the user and do not delete the record from state.